docs/examples/Classifying_Using_HMM.py

```diff
@@ -0,0 +1,232 @@
+
+predictor = HMMPredictor(category_transition)
+
+# %%
+# Updater
+# -------
+from stonesoup.updater.categorical import HMMUpdater
+
+updater = HMMUpdater()
+
+# %%
+# Hypothesiser
+# ------------
+# %%
+# A :class:`~.CategoricalHypothesiser` is used for calculating categorical hypotheses.
+# It utilises the :class:`~.ObservationAccuracy` measure: a multi-dimensional extension of an
+# 'accuracy' score, essentially providing a measure of the similarity between two categorical
+# distributions.
+from stonesoup.hypothesiser.categorical import CategoricalHypothesiser
+
+hypothesiser = CategoricalHypothesiser(predictor=predictor, updater=updater)
+
+# %%
+# Data Associator
+# ---------------
+# We will use a standard :class:`~.GNNWith2DAssignment` data associator.
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+# %%
+# Prior
+# -----
+# As we are tracking in a categorical state space, we should initiate with a categorical state for
+# the prior. Equal probability is given to all 3 of the possible hidden classes that a target
+# might take (the category names are also provided here).
+from stonesoup.types.state import CategoricalState
+
+prior = CategoricalState([1 / 3, 1 / 3, 1 / 3], category_names=hidden_classes)
+
+# %%
+# Initiator
+# ---------
+# For each unassociated detection, a new track will be initiated. In this instance we use a
+# :class:`~.SimpleCategoricalInitiator`, which specifically handles categorical state priors.
+from stonesoup.initiator.categorical import SimpleCategoricalInitiator
+
+initiator = SimpleCategoricalInitiator(prior_state=prior, measurement_model=None)
+
+# %%
+# Deleter
+# -------
+# We can use a standard :class:`~.UpdateTimeStepsDeleter`.
+from stonesoup.deleter.time import UpdateTimeStepsDeleter
+
+deleter = UpdateTimeStepsDeleter(2)
+
+# %%
+# Tracker
+# -------
+# We can use a standard :class:`~.MultiTargetTracker`.
+from stonesoup.tracker.simple import MultiTargetTracker
+
+tracker = MultiTargetTracker(initiator, deleter, measurements, data_associator, updater)
+
+# %%
+# Tracking
+# ^^^^^^^^
+
+tracks = set()
+for time, ctracks in tracker:
+    tracks.update(ctracks)
+
+print(f"Number of tracks: {len(tracks)}")
+for track in tracks:
+    certainty = track.state_vector[np.argmax(track.state_vector)][0] * 100
+    print(f"id: {track.id} -- category: {track.category} -- certainty: {certainty}%")
+    for state in track:
+        meas_string = f"associated measurement: {state.hypothesis.measurement.category}"
+        print(f"{state.timestamp} -- {state.category} -- {meas_string}")
+    print()
+
+# %%
+# Metric
+# ^^^^^^
+# Determining tracking accuracy.
+# In calculating how many targets were classified correctly, only tracks with the highest
+# classification certainty are considered.
+
+excess_tracks = len(tracks) - len(ground_truths)  # target value = 0
+sorted_tracks = sorted(tracks,
```

Comment:
![hmm_output](https://user-images.githubusercontent.com/95758965/146923358-0eef5117-ec42-4191-9489-6def48bc9ccc.PNG)
This was my output from printing some values from here.  The issue I have is that we are being told there is 99.9% classification certainty for the top two tracks, but then 1 of them is wrong.  Obviously there is a possibility that this is random chance, but it happens every few times I run the code (without a seed).  Additionally, it seems strange that in this and other outputs, one track's probabilities are unchanged from the prior.

Reply:
One additional problem I noticed: my understanding is that the probabilities inside `track.state_vector` are in the order bike, car, bus, due to how `hidden_classes` is defined on line 54.  According to this, then, shouldn't `track_classifications` be picking bus twice?  Note this doesn't happen every rerun.  I've tried to work out why this is happening but I'm not sure unfortunately.

Reply:
Good spot. There should be three elements in each set but I'd mistakenly used a set which won't add repeated elements. I'll replace with list.